exchanges/coinbasepro/coinbasepro_websocket.go

```diff
@@ -411,26 +411,29 @@ func (c *CoinbasePro) Subscribe(channelsToSubscribe []stream.ChannelSubscription
 	subscribe := WebsocketSubscribe{
 		Type: "subscribe",
 	}
-
-subscriptions:
+	productIDs := make([]string, 0, len(channelsToSubscribe))
 	for i := range channelsToSubscribe {
 		p := channelsToSubscribe[i].Currency.String()
-		if !common.StringDataCompare(subscribe.ProductIDs, p) && p != "" {
-			subscribe.ProductIDs = append(subscribe.ProductIDs, p)
+		if p != "" && !common.StringDataCompare(subscribe.ProductIDs, p) {
```

Comment:
should compare to `productIDs` as this will keep appending

Reply:
Good catch 🧤 I have done that + added the update to unsubscibes too. The way I tested it was to call unsubscribe once subscriptions were done 🌞